### PR TITLE
SONAR-8648 add organization parameter to api/components/search

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java [hunk 1]
```diff
@@ -77,17 +77,37 @@ public ComponentDto selectOrFailByUuid(DbSession session, String uuid) {
...
-    if (query.getComponentIds() != null && query.getComponentIds().isEmpty()) {
+    return selectByQueryImpl(session, null, query, offset, limit);
+  }
+
+  public List<ComponentDto> selectByQuery(DbSession dbSession, String organizationUuid, ComponentQuery query, int offset, int limit) {
+    requireNonNull(organizationUuid, "organizationUuid can't be null");
+    return selectByQueryImpl(dbSession, organizationUuid, query, offset, limit);
+  }
+
+  private List<ComponentDto> selectByQueryImpl(DbSession session, @Nullable String organizationUuid, ComponentQuery query, int offset, int limit) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: MAJOR Make "selectByQueryImpl" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

File: sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java [hunk 2]
```diff
@@ -77,17 +77,37 @@ public ComponentDto selectOrFailByUuid(DbSession session, String uuid) {
...
-    if (query.getComponentIds() != null && query.getComponentIds().isEmpty()) {
+    return countByQueryImpl(session, null, query);
+  }
+
+  public int countByQuery(DbSession session, String organizationUuid, ComponentQuery query) {
+    requireNonNull(organizationUuid, "organizationUuid can't be null");
+    return countByQueryImpl(session, organizationUuid, query);
+  }
+
+  private int countByQueryImpl(DbSession session, @Nullable String organizationUuid, ComponentQuery query) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: MAJOR Make "countByQueryImpl" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchAction.java [hunk 3]
```diff
@@ -82,6 +87,12 @@ public void define(WebService.NewController context) {
       .setResponseExample(getClass().getResource("search-components-example.json"))
       .setHandler(this);
     action
+      .createParam(PARAM_ORGANIZATION)
+      .setDescription("Organization key")
+      .setRequired(false)
+      .setInternal(true)
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchAction.java]: Missing `setSince("6.3")`

File: sonar-db/src/main/java/org/sonar/db/component/ComponentMapper.java [hunk 4]
```diff
@@ -65,9 +65,9 @@
    */
   int countComponentByOrganizationAndId(@Param("organizationUuid") String organizationUuid, @Param("componentId") long componentId);
 
-  List<ComponentDto> selectByQuery(@Param("query") ComponentQuery query, RowBounds rowBounds);
+  List<ComponentDto> selectByQuery(@Nullable @Param("organizationUuid") String organizationUuid, @Param("query") ComponentQuery query, RowBounds rowBounds);
```
Comment (simonbrandhof) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentMapper.java]: What about moving `organizationUuid` to ComponentQuery ?

File: sonar-db/src/main/java/org/sonar/db/component/ComponentMapper.java [hunk 5]
```diff
@@ -65,9 +65,9 @@
...
 
-  int countByQuery(@Param("query") ComponentQuery query);
+  int countByQuery(@Nullable @Param("organizationUuid") String organizatiionUuid, @Param("query") ComponentQuery query);
```
Comment (simonbrandhof) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentMapper.java]: Small typo in `organizatiionUuid`

File: sonar-db/src/main/java/org/sonar/db/component/ComponentMapper.java (hunk 4 above)
Comment (sns-seb) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentMapper.java]: `ComponentQuery` is used to filter result based on the component's properties.
The organization is on top of that (or totally missing, ie. nullable).

**Pull request merged**